src/wallet/wallet2.cpp

```diff
@@ -1602,6 +1602,53 @@ std::string wallet2::get_subaddress_label(const cryptonote::subaddress_index& in
   return m_subaddress_labels[index.major][index.minor];
 }
 //----------------------------------------------------------------------------------------------------
+void wallet2::scan_tx(const std::set<std::string> txids)
+{
+  // If keys are encrypted, ask for password now to avoid timing leak when finding first output.
+  if (m_ask_password == AskPasswordToDecrypt && !m_unattended && !m_watch_only )//&& !m_multisig_rescan_k)
+  {
+    static critical_section password_lock;
+    CRITICAL_REGION_LOCAL(password_lock);
+    if (!m_encrypt_keys_after_refresh)
+    {
+        boost::optional<epee::wipeable_string> pwd = m_callback->on_get_password("scan_tx");
+        THROW_WALLET_EXCEPTION_IF(!pwd, error::password_needed, tr("Password is needed to scan transactions"));
+        THROW_WALLET_EXCEPTION_IF(!verify_password(*pwd), error::password_needed, tr("Invalid password: password is needed to scan transactions"));
+        decrypt_keys(*pwd);
+        m_encrypt_keys_after_refresh = *pwd;
+    }
+  }
```

Comment:
This should not be needed. Was it in your tests ?

Reply:
Yes, sry, I should remove it and scan_output would ask for password when a first output was encountered.